app/(admin)/admin/leagues/page.tsx

```diff
@@ -1,34 +1,48 @@
 // Copyright (c) Gridiron Survivor.
 // Licensed under the MIT License.
 
-import { JSX } from 'react';
-import { LeagueCard } from '@/components/LeagueCard/LeagueCard';
+'use client';
+import { getAllLeagueEntries } from '@/api/apiFunctions';
+import { getUserLeagues } from '@/utils/utils';
+import { IEntryWithLeague } from '@/components/TableColumns/TableColumns.interface';
+import { JSX, useEffect, useState } from 'react';
+import { leagueColumns } from '@/components/TableColumns/TableColumns';
+import TableData from '@/components/TableData/TableData';
+import { useDataStore } from '@/store/dataStore';
 
 /**
  * Renders the admin page.
  * @returns {JSX.Element} - The rendered Admin Leagues page.
  */
 const AdminLeagues = (): JSX.Element => {
+  const [leaguesData, setLeaguesData] = useState<IEntryWithLeague[]>([]);
+  const { user } = useDataStore((state) => state);
+
+  /**
+   * Get all leagues the user is a part of.
+   */
+  const fetchData = async (): Promise<void> => {
```

Comment:
Q: Is there no need for a `try-catch` block here so when the fetchData fails, it will trigger a toast notification for the user?

Reply:
i deleted it per Mike's insight for now unless Shashi tells me to add it back